spec/recipes/server_spec.rb

```diff
@@ -85,6 +85,41 @@
     include_examples 'allow weak hmacs'
   end
 
+  context 'with permit_tunnel set to true' do
+    cached(:chef_run) do
+      ChefSpec::ServerRunner.new do |node|
+        node.normal['ssh-hardening']['ssh']['server']['permit_tunnel'] = true
+      end.converge(described_recipe)
+    end
+  end
+
+  context 'with permit_tunnel at deafult value of false' do
+    cached(:chef_run) do
+      ChefSpec::ServerRunner.new do |node|
+        node.normal['ssh-hardening']['ssh']['server']['permit_tunnel'] = false
+      end.converge(described_recipe)
+    end
+  end
+
+  context 'with permit_tunnel set to a valid string' do
+    cached(:chef_run) do
+      ChefSpec::ServerRunner.new do |node|
+        node.normal['ssh-hardening']['ssh']['server']['permit_tunnel'] = 'ethernet'
+      end.converge(described_recipe)
+    end
```

Comment:
you miss a test within some contexts, it might be something like:

```ruby
  context 'with permit_tunnel set to a valid string' do
     let(:chef_run) do
       ChefSpec::ServerRunner.new do |node|
         node.normal['ssh-hardening']['ssh']['server']['permit_tunnel'] = 'ethernet'
       end.converge(described_recipe)
     end

     it 'should set PermitTunnel to ethernet' do
       expect(chef_run).to render_file('/etc/ssh/sshd_config').with_content('PermitTunnel ethernet')
     end
  end
```